flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecSink.java

```diff
@@ -120,13 +121,16 @@ public StreamExecSink(
                 rowtimeFieldIndices.add(i);
             }
         }
+
+        final boolean isCollectSink = tableSinkSpec.getTableSink() instanceof CollectDynamicSink;
```

Comment:
Wouldn't it be "cleaner" if an `isCollectSink()` method (or simply`isCollect()`) is introduced to the iface?
Or alternatively, if the method is implemented in a utility class in that package, so that changing the class to `public` just for the purpose of `instanceof` call is avoided?

Reply:
> Wouldn't it be "cleaner" if an isCollectSink() method (or simplyisCollect()) is introduced to the iface?

`DynamicTableSink` is a public API, so we'd be throwing a method into it for internal purposes. Such a method also breaks the abstraction, because you're asking the interface for whether it is a specific implementation.

> Or alternatively, if the method is implemented in a utility class in that package, so that changing the class to public just for the purpose of instanceof call is avoided?

Then you'd just have to make that utility public instead, no?

Reply:
> `DynamicTableSink` is a public API, so we'd be throwing a method into it for internal purposes. Such a method also breaks the abstraction, because you're asking the interface for whether it is a specific implementation.

True, thx!

> Then you'd just have to make that utility public instead, no?

Yep, not insisting, just to me seems a bit cleaner than exposing the class as public just to enable the `instanceof` call.